akka-actor/src/main/scala/akka/io/InetAddressDnsResolver.scala

```diff
@@ -29,41 +29,61 @@ class InetAddressDnsResolver(cache: SimpleDnsCache, config: Config) extends Acto
   // Deprecated JVM property key, keeping for legacy compatibility; replaced by NegativeCachePolicyProp
   private final val NegativeCachePolicyPropFallback = "sun.net.inetaddr.negative.ttl"
 
-  // default values (-1 and 0 are magic numbers, trust them)
-  private final val Forever = -1
-  private final val Never = 0
-  private final val DefaultPositive = 30
+  private final val DefaultPositive = FiniteCache(30.seconds)
 
-  private lazy val cachePolicy: Int = {
-    val n = Try(Security.getProperty(CachePolicyProp).toInt)
+  private lazy val cachePolicy: CachePolicy =
+    Try(Security.getProperty(CachePolicyProp).toInt)
       .orElse(Try(System.getProperty(CachePolicyPropFallback).toInt))
+      .map(parsePolicy)
       .getOrElse(DefaultPositive) // default
-    if (n < 0) Forever else n
-  }
 
-  private lazy val negativeCachePolicy = {
-    val n = Try(Security.getProperty(NegativeCachePolicyProp).toInt)
+  private lazy val negativeCachePolicy: CachePolicy =
+    Try(Security.getProperty(NegativeCachePolicyProp).toInt)
       .orElse(Try(System.getProperty(NegativeCachePolicyPropFallback).toInt))
-      .getOrElse(0) // default
-    if (n < 0) Forever else n
+      .map(parsePolicy)
+      .getOrElse(NeverCache) // default
+
+  private def parsePolicy(n: Int): CachePolicy = {
+    n match {
+      case 0          ⇒ CacheForever
+      case x if x < 0 ⇒ NeverCache
```

Comment:
This is the opposite in than the original.

In the original (above) `n < 0` matches `CacheForever` (i.e `-1`) and `0` matches `NeverCache`.  Why the change?

Reply:
It's a mistake. Another missing test, I guess.

Reply:
Thanks @dwijnand

Reply:
case "infinite" => CacheForever
case "never" => NeverCache